File: php/features.php
<?php
declare(strict_types=1);

// GlobalFlood SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class GlobalFloodFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new GlobalFloodBaseFeature();
            case "test":
                return new GlobalFloodTestFeature();
            default:
                return new GlobalFloodBaseFeature();
        }
    }
}

File: py/features.py
# GlobalFlood SDK feature factory

from feature.base_feature import GlobalFloodBaseFeature
from feature.test_feature import GlobalFloodTestFeature


def _make_feature(name):
    features = {
        "base": lambda: GlobalFloodBaseFeature(),
        "test": lambda: GlobalFloodTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
